Gems/LyShine/Code/Source/UiRenderer.cpp

```diff
@@ -299,6 +299,29 @@ AZ::Matrix4x4 UiRenderer::GetModelViewProjectionMatrix()
     return modelViewProjMat;
 }
 
+#if defined(CARBONATED)
+AZ::Matrix4x4 UiRenderer::GetModelViewProjectionMatrix(const AZ::Vector2& viewportSize)
+{
+    const float viewWidth = viewportSize.GetX();
+    const float viewHeight = viewportSize.GetY();
+
+    float zf = 0.0f;
+    float zn = 1.0f;
+    auto viewportContext = GetViewportContext();
+    if (viewportContext)
+    {
+        const AZ::RHI::Viewport& viewport = viewportContext->GetWindowContext()->GetViewport();
```

Comment:
Please ensure that the feature works with r_renderScale = 0.8 or less. Just by looking at the code it is not easy to derive what viewports / resolutions are in use.

Reply:
It is good, but please also check user input if it goes to the right buttons. I saw in the debugger that user input comes in screen coordinates.